src/settings_doc/main.py

```diff
@@ -61,6 +62,18 @@ def _model_fields_recursive(
                 prefix + field_name + env_nested_delimiter,
                 env_nested_delimiter,
             )
+        elif isclass(model_field.annotation) and issubclass(model_field.annotation, BaseSettings):
+            # There are nested fields that do not require a delimiter to be joined. Generate variable names recursively.
+            submodel_prefix = model_field.annotation.model_config.get("env_prefix", "")
+
+            if not submodel_prefix:
+                submodel_prefix = prefix + field_name + "_"
+
+            yield from _model_fields_recursive(
+                model_field.annotation,
+                submodel_prefix,
+                model_field.annotation.model_config.get("env_nested_delimiter", None),
+            )
```

Comment:
I was a bit worried about this condition and the one above. They seemed non-exhaustive and I didn't quite understand what were the missing cases. Then I found out the following part is actually incorrect:

```python
if not submodel_prefix:
    submodel_prefix = prefix + field_name + "_"
```

which lead me to quite a simplification:

```python
def _model_fields_recursive(
    cls: type[BaseModel], prefix: str, env_nested_delimiter: str | None
) -> Iterator[tuple[str, FieldInfo]]:
    for field_name, model_field in cls.model_fields.items():
        if model_field.validation_alias is not None:
            if isinstance(model_field.validation_alias, str):
                yield model_field.validation_alias, model_field
            else:
                LOGGER.error(f"Unsupported validation alias type '{type(model_field.validation_alias)}'.")
        elif isclass(model_field.annotation) and issubclass(model_field.annotation, BaseModel):
            # There are nested fields and they can be joined by a delimiter. Generate variable names recursively.
            if issubclass(model_field.annotation, BaseSettings):
                yield from _model_fields(model_field.annotation)
            else:  # BaseModel
                yield from _model_fields_recursive(
                    model_field.annotation,
                    prefix + field_name + (env_nested_delimiter or ""),
                    env_nested_delimiter or "",
                )
        else:
            yield prefix + field_name, model_field
```

I tested it locally and it all works.